### PR TITLE
Support Journal files created by Fleet Carrier Update game versions

### PR DESCRIPTION
**Pull request by BafDyce**
At least the events that failed to parse correctly from my journal files from 2020/2021. I added all failing log events (out of ~160k) to relevant test files (one file for each game version i had in my files).

I tried to make as little changes to existing data structures as necessary. Most adaptions (where it made sense) were just providing defaults during the de-serialization.
At least one data structure change also applies to modern game versions (`FSDTargetEvent.remaining_jumps_in_route` from `u8` to `u16`), not sure if it would make sense to directly bump it to `u32`.

Additionally, I also put the *detection* of older Journal files (based on filename) behind a feature gate, as I probably did not have all possible events in my logs and others may have additional events in their logs. (-> reducing likelihood that others are impacted by sudden incompatibilities).

I did not add any new test cases. If you think, I should add some for specific events, please tell me. (Or feel free to make adaptions yourself).

### CODE REVIEW

**Comment by rster2002** [comment]
HI! Thank you so much for creating this PR. I'll go through it in the evening, but it looks very promising.

File: src/modules/state/models/resolvers/live_state_resolver.rs
Comment (rster2002) [inline src/modules/state/models/resolvers/live_state_resolver.rs:1]: This is the result of rust fmt

File: src/modules/logs/content/log_event_content/repair_event.rs [hunk 1]
```diff
@@ -1,18 +1,54 @@
...
+use serde::{de, Deserialize, Deserializer, Serialize};
 
 use crate::modules::ship::ShipModule;
 
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
 #[serde(rename_all = "PascalCase")]
 pub struct RepairEvent {
+    #[serde(alias = "Item", deserialize_with = "string_or_seq_string")]
     pub items: Vec<String>,
     pub cost: u64,
```
Comment (rster2002) [inline src/modules/logs/content/log_event_content/repair_event.rs:12]: @BafDyce do you happen to have an test entry for this change? I couldn't find it in the added log files.

Comment (BafDyce) [reply src/modules/logs/content/log_event_content/repair_event.rs:12]: Sure. Two examples, from a journal file created by `"gameversion":"Fleet Carriers Update - Patch 5", "build":"r241328/r0 "`:

```json
{ "timestamp":"2020-09-21T19:04:44Z", "event":"Repair", "Item":"Paint", "Cost":1 }
{ "timestamp":"2020-09-21T19:04:51Z", "event":"Repair", "Item":"Wear", "Cost":10 }
```

**Comment by rster2002** [comment]
Thanks again! Changes have been merged.

**Comment by BafDyce** [comment]
> Thanks again! Changes have been merged.

Thank you for creating this crate in the first place! :)

**Pull request merged**